eZ/Publish/Core/FieldType/RichText/SearchField.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+/**
+ * This file is part of the eZ Publish Kernel package.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ *
+ * @version //autogentag//
+ */
+namespace eZ\Publish\Core\FieldType\RichText;
+
+use eZ\Publish\SPI\Persistence\Content\Field;
+use eZ\Publish\SPI\Persistence\Content\Type\FieldDefinition;
+use eZ\Publish\SPI\FieldType\Indexable;
+use eZ\Publish\SPI\Search;
+use DOMDocument;
+use DOMNode;
+
+/**
+ * Indexable definition for RichText field type.
+ */
+class SearchField implements Indexable
+{
+    /**
+     * Get index data for field for search backend.
+     *
+     * @param \eZ\Publish\SPI\Persistence\Content\Field $field
+     * @param \eZ\Publish\SPI\Persistence\Content\Type\FieldDefinition $fieldDefinition
+     *
+     * @return \eZ\Publish\SPI\Search\Field[]
+     */
+    public function getIndexData(Field $field, FieldDefinition $fieldDefinition)
+    {
+        $document = new DOMDocument();
+        $document->loadXML($field->value->data);
+
+        return array(
+            new Search\Field(
+                'value',
+                $this->extractShortText($document),
+                new Search\FieldType\StringField()
+            ),
+            new Search\Field(
+                'fulltext',
+                $this->extractText($document->documentElement),
+                new Search\FieldType\FullTextField()
+            ),
+        );
+    }
+
+    /**
+     * Extracts text content of the given $node.
+     *
+     * @param \DOMNode $node
+     *
+     * @return string
+     */
+    private function extractText(DOMNode $node)
+    {
+        $text = '';
+
+        if ($node->childNodes) {
+            foreach ($node->childNodes as $child) {
+                $text .= $this->extractText($child);
+            }
+        } else {
+            $text .= $node->nodeValue . ' ';
+        }
+
+        return $text;
```

Comment:
Couldn't we just do a `strip_tags()` on string representation? :stuck_out_tongue_winking_eye: